[@types/temp] error TS2307: Cannot find module '.'

If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/temp` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @DanielRosenwasser
---
I see the following error when compiling typescript code with `@types/temp@0.8.1`:
```
.../node_modules/@types/temp/index.d.ts(10,23): error TS2307: Cannot find module '.'.
```
This module is being installed as a dependency of `polymer-cli@1.7.0-pre.4`.  I am compiling with typescript version 2.8.1.

The following change seems to correct the problem.  Replace the line:

```
import * as temp from ".";
```
with:
```
import * as temp from "./index";
```

--- a/types/temp/index.d.ts
+++ b/types/temp/index.d.ts
@@ -8,41 +8,41 @@
 import * as fs from "fs";
 
 declare namespace temp {
-  export interface OpenFile {
+  interface OpenFile {
     path: string;
     fd: number;
   }
 
-  export interface Stats {
+  interface Stats {
     files: number;
     dirs: number;
   }
 
-  export interface AffixOptions {
+  interface AffixOptions {
     prefix?: string;
     suffix?: string;
     dir?: string;
   }
 
-  export let dir: string;
+  let dir: string;
 
-  export function track(value?: boolean): typeof temp;
+  function track(value?: boolean): typeof temp;
 
-  export function mkdir(affixes?: string | AffixOptions, callback?: (err: any, dirPath: string) => void): void;
+  function mkdir(affixes?: string | AffixOptions, callback?: (err: any, dirPath: string) => void): void;
 
-  export function mkdirSync(affixes?: string | AffixOptions): string;
+  function mkdirSync(affixes?: string | AffixOptions): string;
 
-  export function open(affixes?: string | AffixOptions, callback?: (err: any, result: OpenFile) => void): void;
+  function open(affixes?: string | AffixOptions, callback?: (err: any, result: OpenFile) => void): void;
 
-  export function openSync(affixes?: string | AffixOptions): OpenFile;
+  function openSync(affixes?: string | AffixOptions): OpenFile;
 
-  export function path(affixes?: string | AffixOptions, defaultPrefix?: string): string;
+  function path(affixes?: string | AffixOptions, defaultPrefix?: string): string;
 
-  export function cleanup(callback?: (result: boolean | Stats) => void): void;
+  function cleanup(callback?: (result: boolean | Stats) => void): void;
 
-  export function cleanupSync(): boolean | Stats;
+  function cleanupSync(): boolean | Stats;
 
-  export function createWriteStream(affixes?: string | AffixOptions): fs.WriteStream;
+  function createWriteStream(affixes?: string | AffixOptions): fs.WriteStream;
 }
 
 export = temp;

--- a/types/temp/index.d.ts
+++ b/types/temp/index.d.ts
@@ -5,41 +5,44 @@
 
 /// <reference types="node" />
 
-import * as temp from ".";
 import * as fs from "fs";
 
-export interface OpenFile {
-  path: string;
-  fd: number;
+declare namespace temp {
+  export interface OpenFile {
+    path: string;
+    fd: number;
+  }
+
+  export interface Stats {
+    files: number;
+    dirs: number;
+  }
+
+  export interface AffixOptions {
+    prefix?: string;
+    suffix?: string;
+    dir?: string;
+  }
+
+  export let dir: string;
+
+  export function track(value?: boolean): typeof temp;
+
+  export function mkdir(affixes?: string | AffixOptions, callback?: (err: any, dirPath: string) => void): void;
+
+  export function mkdirSync(affixes?: string | AffixOptions): string;
+
+  export function open(affixes?: string | AffixOptions, callback?: (err: any, result: OpenFile) => void): void;
+
+  export function openSync(affixes?: string | AffixOptions): OpenFile;
+
+  export function path(affixes?: string | AffixOptions, defaultPrefix?: string): string;
+
+  export function cleanup(callback?: (result: boolean | Stats) => void): void;
+
+  export function cleanupSync(): boolean | Stats;
+
+  export function createWriteStream(affixes?: string | AffixOptions): fs.WriteStream;
 }
 
-export interface Stats {
-  files: number;
-  dirs: number;
-}
-
-export interface AffixOptions {
-  prefix?: string;
-  suffix?: string;
-  dir?: string;
-}
-
-export let dir: string;
-
-export function track(value?: boolean): typeof temp;
-
-export function mkdir(affixes?: string | AffixOptions, callback?: (err: any, dirPath: string) => void): void;
-
-export function mkdirSync(affixes?: string | AffixOptions): string;
-
-export function open(affixes?: string | AffixOptions, callback?: (err: any, result: OpenFile) => void): void;
-
-export function openSync(affixes?: string | AffixOptions): OpenFile;
-
-export function path(affixes?: string | AffixOptions, defaultPrefix?: string): string;
-
-export function cleanup(callback?: (result: boolean | Stats) => void): void;
-
-export function cleanupSync(): boolean | Stats;
-
-export function createWriteStream(affixes?: string | AffixOptions): fs.WriteStream;
+export = temp;

--- a/types/temp/tslint.json
+++ b/types/temp/tslint.json
@@ -1,5 +1,6 @@
 {
     "extends": "dtslint/dt.json",
     "rules": {
+        "export-just-namespace": false
     }
 }

